discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -41,7 +44,7 @@ def refresh_user_ids(
     redis, db, token_contract, delegate_manager_contract, staking_contract, eth_web3
 ):
     # List users in Redis set, balances decoded as strings
-    redis_user_ids = redis.smembers(REDIS_PREFIX)
+    redis_user_ids = redis.smembers(REDIS_USER_BALANCE_REFRESH_KEY)
```

Comment:
do we still need the code underneath for 'needs_refresh' with this change @piazzatron?

Reply:
we do not @hareeshnagaraj, good callout. saliou and i are gonna sync about this stuff in an hour

Reply:
let's keep @jowlee in the loop here too since his work also touches this path